### PR TITLE
Dasharo UEFI Secure Boot support for OVMF

### PR DESCRIPTION
**Pull request by pietrushnic**

### CODE REVIEW

File: OvmfPkg/PlatformPei/Platform.c [hunk 1]
```diff
@@ -452,13 +452,71 @@ MiscInitialization (
...
+    UINT16 Expected;
+
+    Expected =
+      (UINT16) (((UINTN) FwVolHeader->Checksum + 0x10000 - Checksum) & 0xffff);
+
+    DEBUG ((EFI_D_INFO, "FV@%p Checksum is 0x%x, expected 0x%x\n",
+            FwVolHeader, FwVolHeader->Checksum, Expected));
+    return EFI_NOT_FOUND;
+  }
+
```
Comment (miczyg1) [inline OvmfPkg/PlatformPei/Platform.c:505]: This is only a part of the verification. Following the FVH there is also NVAR storage header which should be validated too.

File: OvmfPkg/PlatformPei/Platform.c [hunk 2]
```diff
@@ -470,6 +528,20 @@ BootModeInitialization (
 
   Status = PeiServicesInstallPpi (mPpiBootMode);
   ASSERT_EFI_ERROR (Status);
+
+  //
+  // Set BootMode based on FV header state
+  //
+  Status = ValidateFvHeader (FwVolHeader);
```
Comment (miczyg1) [inline OvmfPkg/PlatformPei/Platform.c:535]: This should probably be put before line `if (CmosRead8 (0xF) == 0xFE) {` and use the variable mBootMode:

```C
  //
  // Set BootMode based on FV header state
  //
  Status = ValidateFvHeader (FwVolHeader);

  if (EFI_ERROR (Status)) {
    mBootMode = BOOT_WITH_DEFAULT_SETTINGS;
    DEBUG ((DEBUG_INFO, "BootMode: Boot with default settings\n"));
  } else {
    mBootMode = BOOT_ASSUMING_NO_CONFIGURATION_CHANGES;
    DEBUG ((DEBUG_INFO, "BootMode: Boot boot assuming no configuration changes\n"));
```

**Review by miczyg1 — CHANGES_REQUESTED** [review]

**Comment by pietrushnic** [comment]
@miczyg1 These commits are no longer valid since your new code implementing UEFI Secure Boot works out of the box. No modification is needed. I continue to work on UEFI Secure Boot as part of #106 and will push modifications there if needed.

@macpijan cc

**Comment by macpijan** [comment]
@pietrushnic Have you checked if clearing/resetting keys also works as expected?

**Comment by pietrushnic** [comment]
@macpijan I didn't get too far. I can see that I can enable/disable secure boot in Custom Mode after resetting keys (aka provisioning default one). The next step would be to check if I can install OS in UEFI Secure Boot-aware way.

**Pull request closed without merging**